src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -1071,6 +1059,37 @@ public boolean declaredInCurrentClass(MethodCallExpr method) {
     return false;
   }
 
+  /**
+   * Resolves symbols for a given ClassOrInterfaceType instance, including its type variables if
+   * present.
+   *
+   * @param typeExpr The ClassOrInterfaceType instance to resolve symbols for.
+   */
+  private void solvedSymbolsForTypes(ClassOrInterfaceType typeExpr, boolean isAnInterface) {
+    Optional<NodeList<Type>> typeArguments = typeExpr.getTypeArguments();
+    UnsolvedClassOrInterface classToUpdate;
+    int numberOfArguments = 0;
+    String typeRawName = typeExpr.getElementType().asString();
+    if (typeArguments.isPresent()) {
+      numberOfArguments = typeArguments.get().size();
+      // without any type argument
+      typeRawName = typeRawName.substring(0, typeRawName.indexOf("<"));
+    }
+    if (isAClassPath(typeRawName)) {
+      String packageName = typeRawName.substring(0, typeRawName.lastIndexOf("."));
+      String className = typeRawName.substring(typeRawName.lastIndexOf(".") + 1);
+      classToUpdate = new UnsolvedClassOrInterface(className, packageName);
+    } else {
+      String packageName = classAndPackageMap.getOrDefault(typeRawName, currentPackage);
+      classToUpdate = new UnsolvedClassOrInterface(typeRawName, packageName);
+    }
+    classToUpdate.setNumberOfTypeVariables(numberOfArguments);
+    if (isAnInterface) {
+      classToUpdate.isAnInterface = true;
```

Comment:
Why can't this be set by the constructor calls on lines 1081 and 1084? If you did that, you wouldn't need to make this field non-final or public, I think, so that would be worthwhile.

Reply:
Professor,

I am not sure about the issue here. It seems to me that implementing a setter method for a private field is not different from simply making that field public.

Could you please provide clarification on this matter?

Reply:
> It seems to me that implementing a setter method for a private field is not different from simply making that field public.

You wouldn't need a public setter for the field if the only assignment is in the constructor, which is what I'm suggesting. Ideally, fields shouldn't be writable outside the enclosing class at all: that is, a class should control writes to its own fields. Logically this makes sense, because the field should be _encapsulated_ within the class if it logically belongs to that class.

Reply:
Happy to explain this more in our meeting later today if this explanation doesn't work.